src/ansible_compat/runtime.py

```diff
@@ -380,18 +393,21 @@ def _prepare_ansible_paths(self) -> None:
             (roles_path, "roles", True),
         ]
 
-        if self.isolated:
-            alterations_list.extend(
-                [
-                    (roles_path, f"{self.cache_dir}/roles", False),
-                    (library_paths, f"{self.cache_dir}/modules", False),
-                    (collections_path, f"{self.cache_dir}/collections", False),
-                ]
-            )
+        alterations_list.extend(
+            [
+                (roles_path, f"{self.cache_dir}/roles", False),
+                (library_paths, f"{self.cache_dir}/modules", False),
+                (collections_path, f"{self.cache_dir}/collections", False),
+            ]
+            if self.isolated
+            else []
+        )
 
         for path_list, path, must_be_present in alterations_list:
-            if must_be_present and not os.path.exists(path):
-                continue
+            if not os.path.exists(path):
+                if must_be_present:
+                    continue
+                os.makedirs(path, exist_ok=True)
```

Comment:
Why does this need `exist_ok`?

Reply:
Destination folders may already exist, especially when using the library with isolated=False.